strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/MavenArtifactGenerator.java

```diff
@@ -163,17 +160,17 @@ protected void createMetadata(Metadata metadata, String metadataPath)
             // noinspection ResultOfMethodCallIgnored
             metadataFile.getParentFile().mkdirs();
 
-            os = new MultipleDigestOutputStream(metadataFile, newOutputStream(metadataFile));
-            writer = WriterFactory.newXmlWriter(os);
-            MetadataXpp3Writer mappingWriter = new MetadataXpp3Writer();
-            mappingWriter.write(writer, metadata);
+            try (OutputStream os = new MultipleDigestOutputStream(metadataFile, newOutputStream(metadataFile)))
+            {
+                Writer writer = WriterFactory.newXmlWriter(os);
+                MetadataXpp3Writer mappingWriter = new MetadataXpp3Writer();
+                mappingWriter.write(writer, metadata);
 
-            os.flush();
+                os.flush();
+            }
         }
         finally
         {
-            ResourceCloser.close(os, logger);
```

Comment:
Am I missing something here? Where's the `close()` happening?

Reply:
this is [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) statement, it closes all that declared in `try`.